crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3494,14 +3494,20 @@ impl<'db> TypeInferenceBuilder<'db> {
                 Type::Todo
             }
 
-            // TODO PEP-604 unions
             ast::Expr::BinOp(binary) => {
-                self.infer_binary_expression(binary);
+                #[allow(clippy::single_match_else)]
```

Comment:
using `if`/`let` might be okay here. I don't think there are any other binary operations that are legal in type annotations right now, so it's unlikely we'll ever add any other branches in the medium term. In the long term, intersection types might eventually get standardised, which would mean `str & int` would become a legal annotation... but I think that's still a long way off for now.

Reply:
I personally still find this clearer given the pattern-matching style of the file.

Reply:
Fair. I'm not a massive fan of this Clippy rule in general TBH -- it feels like it has too many false positives for my liking